src/actions/arcActions.ts

```diff
@@ -654,9 +655,9 @@ export function stakeProposal(daoAvatarAddress: string, proposalId: string, pred
       const stakingToken = await StandardToken.at(await votingMachineInstance.contract.stakingToken());
       const balance = await stakingToken.balanceOf(currentAccountAddress);
 
-      const amount = web3.toWei(stake, "ether");
-      if (amount < minimumStakingFee) { throw new Error(`Staked less than the minimum: ${minimumStakingFee}!`); }
-      if (amount > balance) { throw new Error(`Staked more than than the balance: ${balance}!`); }
+      const amount = new BigNumber(Util.toWei(stake));
+      if (amount.lt(minimumStakingFee)) { throw new Error(`Staked less than the minimum: ${Util.fromWei(minimumStakingFee)}!`); }
```

Comment:
In these computations using `.lt` there is no need to cast to a number at all.  I think it would be better that `Utils.fromWei` return a `BigNumber` and let the caller decide whether it needs to cast the result to a number.  (And when doing so, best to use `.toNumber()` rather than `Number()`).

Reply:
Thought about this, this would also have the benefit of making `toWei`/`fromWei` inverses of each other.
But returning a string is more convenient for debugging, e.g. in the `redux-devtools` you'll see the number as a string instead of this weird `BigNumber` representation.
(Also I don't know if there parts in the code which assume it's a string, and this would screw them up)

Reply:
Your choice, but I think it merits a refactoring, at an appropriate time.  In the long run I think it'll be a better design.

Reply:
added a todo: https://github.com/daostack/alchemy/pull/44/commits/b5df9a0f38f4b860160736e6baafce21f29c75b8